DEVELOPER_WORKFLOW.md

```diff
@@ -0,0 +1,191 @@
+
+### Test Pass
+1. Main purpose of test pass is to look for regressions – not if new features work properly.
+2. **<Is this thorough enough?>**
+3. Run all functional tests.
+   1. Test localhost deploy functional tests
+      1. Run functional test scene in unity test bed
+      2. using the functional test menu in synchronization scene in testbed. Go in and out of each test twice. Keep an eye on the synchronized instance and make sure both work
+4. Deploy functional test temporarily to openode/Heroku/azure/ngrok and test 
+5. Verify functional tests runs on Android.
+6. Run all samples
+   1. **TODO: insert link to instructions for running with local NPM package instead of public**
+7. If problems are found, fix and restart test pass
+
+### Publish NPM Packages
+1. Note that we always publish all packages with each release, with version numbers in sync
+2. Check out sdk\v1.[minor].n 
+3. Git reset –hard
+4. git clean -fdx
+5. npm install
+6. npm run build
+7. lerna publish –dist-tag <either next or latest>
+   1. when it asks for version number, use 0.[minor].[patch], where minor comes from branch, and patch version is 1+the last published NPM package for that minor version (to find last published version see [MRE version page on NPMJS](https://www.npmjs.com/package/@microsoft/mixed-reality-extension-sdk?activeTab=versions)
+8. For each updated package.json version number, notepad-update the version number in every corresponding package-lock.json in both sdk and samples repo. Lerna doesn’t do this (but should).
+9. Git commit package.json and package-lock.json changes to sdk\v1.[minor].n
+10. npm run build-docs (to regenerate documentation)
+11. commit documentation changes to sdk\v1.[minor].n
+12. in samples\v1.[minor].n, for each of the samples update the patch version:
+    1. npm update @microsoft/mixed-reality-extension-sdk@0.[minor].[patch]
+    2. npm install
+    3. npm run build
+    4. verify it loads
+    5. git commit package.json and package-lock.json changes to samples\v1.[minor].n
+
+### Update Master Branches
+1. Publish docs from that branch to ??? github pages
+2. Gather the differences from each repo (to see what we have added): git log --oneline origin/master..origin/unity/v0.[minor].n
+3. git reset hard unity/master to unity/v0.[minor].n
+4. git reset hard sdk/master to sdk/v0.[minor].n
+5. git reset hard samples/master to samples/v0.[minor].n
+6. redeploy samples and functional test MREs
+7. Slack: Make announcement to #announcements channel, share announcement on #general channel.
+8. Twitter: Announce new features
+9. Teams: announce on Altspace Community Support->SDK or General
+10. go to [the MRE Github Issues page](https://github.com/Microsoft/mixed-reality-extension-sdk/projects/1) and move released issues from Fixed to Released
+
+### Build Unity DLLs
+1. It builds on server at every commit 
```

Comment:
on every commit to dev branch only. Builds from other branches must be queued manually.